flink-clients/src/main/java/org/apache/flink/client/cli/ProgramOptions.java

```diff
@@ -152,6 +112,26 @@ public ProgramOptions(CommandLine line) throws CliArgsException {
 		this.savepointSettings = CliFrontendParser.createSavepointRestoreSettings(line);
 	}
 
+	protected String[] extractProgramArgs(CommandLine line) {
+		String[] args = line.hasOption(ARGS_OPTION.getOpt()) ?
+			line.getOptionValues(ARGS_OPTION.getOpt()) :
+			line.getArgs();
+
+		if (args.length > 0 && !line.hasOption(JAR_OPTION.getOpt())) {
+			jarFilePath = args[0];
+			args = Arrays.copyOfRange(args, 1, args.length);
+		}
+
+		return args;
+	}
+
+	public void validate() throws CliArgsException {
+		// Java program should be specified a JAR file
+		if (getJarFilePath() == null) {
+			throw new CliArgsException("Java program should be specified a JAR file.");
```

Comment:
I think that's not always necessary, if the user jar is in the lib folder it's enough to specify an entrypoint Class.

Reply:
Yes, either of entrypoint class and jar file is enough. But before this PR, we only check whether ProgramOptions.getJarFilePath() is null in `CliFrontend`, so I kept the original logic in the validate method. It is ok for me to change this validate logic in this PR. What do you think?